pkg/endpoint/endpoint.go

```diff
@@ -1797,3 +1802,71 @@ func (e *Endpoint) identityLabelsChanged(owner Owner, myChangeRev int) error {
 
 	return nil
 }
+
+// setPolicyRevision sets the policy wantedRev with the given revision.
+func (e *Endpoint) setPolicyRevision(rev uint64) {
+	e.policyRevision = rev
+	k := 0
+	for i, w := range e.policyRevisionSignals {
+		select {
+		case <-w.ctx.Done():
+			close(w.ch)
```

Comment:
You also need to handle removing the policySignal struct from the list in this case. Maybe make a function variable before the loop and defer it per thing you need to delete?